test/metabase/lib/convert_test.cljc

```diff
@@ -427,7 +386,8 @@
     {:database 1
      :type     :query
      :query    {:source-table 224
-                :expressions {"a" 1}}}
+                :expressions {"a" 1}
+                :expression-idents {"a" (u/generate-nano-id)}}}
```

Comment:
They're opaque, random strings. Most tests should use `mt/mbql-query` and friends, which will (1) automatically generate `:expression-idents` etc.; and (2) has metadata that makes `=?` ignore the generated idents on the `expected` side.

Reply:
To amplify that first statement: these are opaque, unique strings. We currently happen to be using NanoIDs when we need a new `:ident` _ex nihilo_ for a newly created aggregation/breakout/expression/join, but nothing should be **reading** these idents.

The only legal operations on these idents are:
- matching by `=` (including as map keys), and
- concatenating other strings to the ends
    - eg. to attach a fragment for a join to each joined column's ident

Reply:
@bshepherdson my comment was badly worded, but by "painful" I was thinking of the potential need to migrate stored ids if we changed the format (but maybe even that is not necessary).

Really I just meant "unlikely" rather than "painful"